lib/3scale_toolbox/cli/error_handler.rb

```diff
@@ -0,0 +1,133 @@
+module ThreeScaleToolbox::CLI
+  # Catches errors and prints nice diagnostic messages
+  class ErrorHandler
+    def self.error_watchdog(exit_on_error: true)
+      new.error_watchdog(exit_on_error: exit_on_error) { yield }
+    end
+
+    def error_watchdog(exit_on_error:)
+      # Set exit handler
+      %w[INT TERM].each do |signal|
+        Signal.trap(signal) do
+          puts
+          exit!(0)
+        end
+      end
+
+      # Set stack trace dump handler
+      if !defined?(RUBY_ENGINE) || RUBY_ENGINE != 'jruby'
+        begin
+          Signal.trap('USR1') do
+            puts 'Caught USR1; dumping a stack trace'
+            puts caller.map { |i| "  #{i}" }.join("\n")
+          end
+        rescue ArgumentError
+        end
+      end
+
+      # Run
+      yield
+    rescue StandardError, ScriptError => e
+      handle_error(e, exit_on_error: exit_on_error)
+    end
+
+    private
+
+    def handle_error(error, exit_on_error:)
+      if expected_error?(error)
+        warn
+        warn "Error: #{error.message}"
+      else
+        print_error(error)
+      end
+      exit(1) if exit_on_error
```

Comment:
Seems like the better place for this is the main CLI module. This flag is being passed across several methods in the end to just call exit.